### ISSUE TITLE
Template variable list items have no padding

### ISSUE THREAD

**Issue by ohmakai** [grafana/grafana#65953]
**What happened**:
While upgrading to v9.4, I noticed some UI changes that seem like bugs. Specifically the way template variable list items are padded and also the way table headers are rendered (thought that has been captured in [#65772](https://github.com/grafana/grafana/issues/65772) )

**What you expected to happen**:
No change to template variable list item spacing

**How to reproduce it (as minimally and precisely as possible)**:
- Using Grafana v9.4.x, create a new dashboard
- Create a new template variable (can be a query, custom, etc.)
- View dashboard
- Click on template variable dropdown

**Anything else we need to know?**:
Looks like there is a compiled classname that obfuscates the original padding: `.css-1nqk4pg`

**Environment**:
- Grafana version: 9.4.x
- Data source type & version: any
- OS Grafana is installed on: Linux (Using Docker container)
- User OS & Browser: MacOS + Chrome
- Grafana plugins: any
![Screenshot 2023-04-04 at 10 36 27 AM](https://user-images.githubusercontent.com/477793/229875989-94d7bd21-7470-4d8d-8002-fba363cb303f.png)

**Comment by ohmakai** [comment]
Looking into it a bit, I noticed that the model changed for template variables - it went from using an anchor to using a button. 
before v9.3:
```
<ul class="variable-options-column css-1oo66qz" aria-label="Variable options" id="options-service">
        <li>
            <a role="checkbox" aria-checked="false" class="variable-option pointer">
                <span class="variable-option-icon"></span>
                <span data-testid="data-testid Dashboard template variables Variable Value DropDown option text 3p-webhook-event-replay">Some variable option</span>
            </a>
        </li>
</ul>
```

on or after v9.3:
```
<ul class="variable-options-column css-1oo66qz" aria-label="Variable options" id="options-service" theme="[object Object]">
    <li>
        <button role="checkbox" type="button" aria-checked="false" class="variable-option pointer css-1nqk4pg css-bv3w3n">
            <span class="variable-option-icon"></span>
            <span data-testid="data-testid Dashboard template variables Variable Value DropDown option text 3p-webhook-event-replay">Some variable option</span>
        </button>
    </li>
</ul>
```

**Comment by natellium** [comment]
Thanks for reporting! Was able to reproduce in 9.4.x but it will be fixed in 9.5.x. Closing this issue, please ping us if the issue persists and we'll reopen it :)
